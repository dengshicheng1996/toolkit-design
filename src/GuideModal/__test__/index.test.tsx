import * as React from 'react'
import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import BasicGuideModal from '../demo/basic'

describe('GuideModal', () => {
  it('default snapshot', () => {
    const component = mount(<BasicGuideModal data-y="111" />)

    expect(toJSON(component)).toMatchSnapshot()
  })
})
