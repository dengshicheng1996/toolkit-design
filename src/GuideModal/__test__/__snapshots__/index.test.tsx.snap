// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuideModal default snapshot 1`] = `
<_default
  data-y="111"
>
  <_default>
    <div
      className="box"
    >
      <div
        className="layerWrap"
      >
        <img
          className="layer"
          src="https://img.alicdn.com/tfs/TB1AcmptQL0gK0jSZFtXXXQCXXa-1500-800.png"
        />
        <div
          className="group"
        >
          <div
            className="container"
          >
            <img
              className="icon"
              src="https://img.alicdn.com/tfs/TB11jittUY1gK0jSZFCXXcwqXXa-40-68.png"
            />
            <span
              className="return"
            >
              返回
            </span>
            <span
              className="title"
            >
              天天走路得保障
            </span>
          </div>
          <span
            className="rules"
          >
            规则
          </span>
        </div>
        <img
          className="xingzhuang"
          src="https://img.alicdn.com/tfs/TB12.9rtO_1gK0jSZFqXXcpaXXa-30-30.png"
        />
        <div
          className="block_3"
          id="headerImgArea"
        >
          <img
            className="iconduty"
            src="https://img.alicdn.com/tfs/TB1GR5stKL2gK0jSZFmXXc7iXXa-80-62.png"
          />
          <div
            className="block_2"
          >
            <div
              className="outer"
            >
              <img
                className="QQkongjian"
                src="https://img.alicdn.com/tfs/TB182SwtRr0gK0jSZFnXXbRRXXa-42-42.png"
              />
              <img
                className="meiyijia"
                src="https://img.alicdn.com/tfs/TB12hqqtHr1gK0jSZR0XXbP8XXa-42-42.png"
              />
              <img
                className="info"
                src="https://img.alicdn.com/tfs/TB1mqCrtQL0gK0jSZFAXXcA9pXa-42-40.png"
              />
              <img
                className="wodeCRM"
                src="https://img.alicdn.com/tfs/TB1lOSqtSf2gK0jSZFPXXXsopXa-42-42.png"
              />
              <img
                className="iconJianpan"
                src="https://img.alicdn.com/tfs/TB1WE9rtO_1gK0jSZFqXXcpaXXa-42-42.png"
              />
              <div
                className="guanbiWrap"
              >
                <img
                  className="yishuxiuxi"
                  src="https://img.alicdn.com/tfs/TB1LM1ttND1gK0jSZFKXXcJrVXa-36-38.png"
                />
                <img
                  className="guanbi"
                  src="https://img.alicdn.com/tfs/TB14GurtQL0gK0jSZFAXXcA9pXa-36-38.png"
                />
              </div>
            </div>
            <div
              className="block"
            >
              <img
                className="img"
                src="https://img.alicdn.com/tfs/TB1_qurtQL0gK0jSZFAXXcA9pXa-14-16.png"
              />
              <img
                className="dot"
                src="https://img.alicdn.com/tfs/TB1yCiutKH2gK0jSZJnXXaT1FXa-10-16.png"
              />
              <img
                className="verticalLine"
                src="https://img.alicdn.com/tfs/TB1iWyotFY7gK0jSZKzXXaikpXa-4-16.png"
              />
              <img
                className="dot_2"
                src="https://img.alicdn.com/tfs/TB1MWGutHj1gK0jSZFOXXc7GpXa-12-16.png"
              />
              <img
                className="img_2"
                src="https://img.alicdn.com/tfs/TB1_qurtQL0gK0jSZFAXXcA9pXa-14-16.png"
              />
              <img
                className="image"
                src="https://img.alicdn.com/tfs/TB1Vo5rtO_1gK0jSZFqXXcpaXXa-12-16.png"
              />
              <img
                className="dot_3"
                src="https://img.alicdn.com/tfs/TB1VkeotLb2gK0jSZK9XXaEgFXa-12-16.png"
              />
              <img
                className="dot_4"
                src="https://img.alicdn.com/tfs/TB1MWGutHj1gK0jSZFOXXc7GpXa-12-16.png"
              />
              <img
                className="dot_5"
                src="https://img.alicdn.com/tfs/TB1LjaotUz1gK0jSZLeXXb9kVXa-12-16.png"
              />
              <img
                className="image_2"
                src="https://img.alicdn.com/tfs/TB1MN1qtUT1gK0jSZFhXXaAtVXa-12-16.png"
              />
              <img
                className="dot_6"
                src="https://img.alicdn.com/tfs/TB1RN1qtUT1gK0jSZFhXXaAtVXa-12-16.png"
              />
              <img
                className="dot_7"
                src="https://img.alicdn.com/tfs/TB1VkeotLb2gK0jSZK9XXaEgFXa-12-16.png"
              />
            </div>
          </div>
          <div
            className="group_2"
          >
            <img
              className="yingyong"
              src="https://img.alicdn.com/tfs/TB1KH1qtHr1gK0jSZFDXXb9yVXa-48-14.png"
            />
            <img
              className="erchuangkou"
              src="https://img.alicdn.com/tfs/TB1YYqqtNn1gK0jSZKPXXXvUXXa-48-34.png"
            />
            <img
              className="qukuaiPeixun"
              src="https://img.alicdn.com/tfs/TB1OSeutKH2gK0jSZJnXXaT1FXa-54-48.png"
            />
            <img
              className="shengjiCopy"
              src="https://img.alicdn.com/tfs/TB1OdSutQT2gK0jSZFkXXcIQFXa-22-52.png"
            />
            <img
              className="logo"
              src="https://img.alicdn.com/tfs/TB1yWmptKL2gK0jSZPhXXahvXXa-254-70.png"
            />
            <img
              className="kehuguanli"
              src="https://img.alicdn.com/tfs/TB1PKaotFT7gK0jSZFpXXaTkpXa-60-70.png"
            />
            <img
              className="gen"
              src="https://img.alicdn.com/tfs/TB1RlWstKL2gK0jSZFmXXc7iXXa-60-70.png"
            />
            <img
              className="mobileFill"
              src="https://img.alicdn.com/tfs/TB1LWuotFY7gK0jSZKzXXaikpXa-28-48.png"
            />
          </div>
        </div>
      </div>
      <div
        className="bd"
        id="stepCardArea"
      >
        <div
          className="titleWrap"
        >
          <span
            className="title_2"
          >
            可领免费健康保额
          </span>
        </div>
        <div
          className="block_4"
        >
          <span
            className="count"
          >
            8.20
          </span>
          <span
            className="yuan"
          >
            元
          </span>
        </div>
        <div
          className="group_3"
        >
          <div
            className="empty"
          />
        </div>
        <img
          className="iconTest"
          src="https://img.alicdn.com/tfs/TB1vxSqtUT1gK0jSZFhXXaAtVXa-76-76.png"
        />
        <div
          className="block_5"
        >
          <span
            className="stepToday"
          >
            今日4120步
          </span>
        </div>
        <div
          className="freeCollectionOuter"
        >
          <div
            className="freeCollectionWrap"
          >
            <span
              className="freeCollection"
            >
              免费领取
            </span>
          </div>
        </div>
        <div
          className="titleWrap_2"
        >
          <span
            className="title_3"
          >
            每天领1次，步数越多领得保额越多～
          </span>
        </div>
      </div>
      <div
        className="ft"
      >
        <span
          className="title_4"
        >
          邀好友领 你得额外保额奖励
        </span>
        <span
          className="title_5"
        >
          好友当日领取，保额奖励自动到账
        </span>
        <div
          className="group_4"
        >
          <div
            className="block_6"
          >
            <div
              className="color"
            />
            <span
              className="desc"
            >
              天边一朵云领取，你获得20元保额奖励
            </span>
            <img
              className="v"
              src="https://img.alicdn.com/tfs/TB1P8SstKL2gK0jSZFmXXc7iXXa-30-52.png"
            />
          </div>
        </div>
        <div
          className="block_7"
        >
          <div
            className="titleWrap_3"
          >
            <img
              className="pic"
              src="https://img.alicdn.com/tfs/TB1AYqqtNn1gK0jSZKPXXXvUXXa-160-160.png"
            />
            <span
              className="title_6"
            >
              天边一朵云
            </span>
          </div>
          <div
            className="shoutCollarWrap"
          >
            <span
              className="shoutCollar"
            >
              喊TA领
            </span>
          </div>
        </div>
        <div
          className="group_5"
        >
          <div
            className="div"
          />
        </div>
        <div
          className="block_8"
        >
          <div
            className="titleWrap_4"
          >
            <img
              className="img_3"
              src="https://img.alicdn.com/tfs/TB1AYqqtNn1gK0jSZKPXXXvUXXa-160-160.png"
            />
            <span
              className="title_7"
            >
              天边一朵云
            </span>
          </div>
          <div
            className="shoutCollarWrap_2"
          >
            <span
              className="shoutCollar_2"
            >
              喊TA领
            </span>
          </div>
        </div>
        <div
          className="group_6"
        >
          <div
            className="empty_2"
          />
        </div>
      </div>
      <NewUserGuide>
        <div
          className="guide"
        >
          <GuideModal
            className="basic-guide-modal"
            fatigue={100000}
            onChange={[Function]}
            onMaskPress={[Function]}
            outStepEleIds={
              Array [
                "headerImgArea",
                "stepCardArea",
              ]
            }
            showSkip={true}
            storageKey="DEFAULT_GUIDE_MODEL"
          >
            <ErrorBoundary
              onError={[Function]}
            >
              <GuideModal
                className="basic-guide-modal"
                fatigue={100000}
                onChange={[Function]}
                onMaskPress={[Function]}
                outStepEleIds={
                  Array [
                    "headerImgArea",
                    "stepCardArea",
                  ]
                }
                showSkip={true}
                storageKey="DEFAULT_GUIDE_MODEL"
              >
                <Portal
                  containerInfo={
                    <div
                      id="amd-guide-modal"
                    >
                      <div
                        class="amd-guide-modal basic-guide-modal"
                      >
                        <div
                          class=""
                        >
                          <div
                            class="top-img"
                          >
                            <img
                              src="https://gw.alipayobjects.com/mdn/rms_2cf5b1/afts/img/A*DRWiTa03CvwAAAAAAAAAAABkARQnAQ"
                            />
                          </div>
                          <div
                            class="top guide-desc"
                          >
                            <div>
                              <span
                                class="blue"
                              >
                                运动
                              </span>
                              联合
                              <span
                                class="yellow"
                              >
                                相互宝
                              </span>
                            </div>
                            <div>
                              送你健康保障福利
                            </div>
                          </div>
                          <div
                            class="amd-guide-modal-next-btn-wrap"
                          >
                            <div
                              aria-disabled="false"
                              class="amd-button amd-button-default amd-guide-modal-next-btn amd-button-capsule amd-button-capsule-md"
                              role="button"
                            >
                              <span
                                style="opacity: 1;"
                              >
                                跳过
                                
                              </span>
                            </div>
                            <div
                              aria-disabled="false"
                              class="amd-button amd-button-default amd-button-capsule amd-button-capsule-md"
                              role="button"
                            >
                              <span
                                style="opacity: 1;"
                              >
                                下一步
                                
                              </span>
                            </div>
                          </div>
                        </div>
                        <div
                          class="amd-guide-modal-step-hide"
                        >
                          <div
                            class="bottom-img"
                          >
                            <img
                              src="https://gw.alipayobjects.com/mdn/rms_2cf5b1/afts/img/A*DRWiTa03CvwAAAAAAAAAAABkARQnAQ"
                            />
                          </div>
                          <div
                            class="bottom guide-desc"
                          >
                            <div>
                              每天走路领保额，每天最高领
                              <span
                                class="yellow"
                              >
                                50元
                              </span>
                            </div>
                            <div>
                              保障100种重大疾病
                            </div>
                          </div>
                          <div
                            class="amd-guide-modal-next-btn-wrap"
                          >
                            <div
                              aria-disabled="false"
                              class="amd-button amd-button-default amd-guide-modal-next-btn amd-button-capsule amd-button-capsule-md"
                              role="button"
                            >
                              <span
                                style="opacity: 1;"
                              >
                                知道了
                                
                              </span>
                            </div>
                          </div>
                        </div>
                        <div
                          class="amd-guide-modal-indicator-area"
                        >
                          <span
                            class="amd-guide-modal-indicator amd-guide-modal-indicator-active"
                          />
                          <span
                            class="amd-guide-modal-indicator"
                          />
                        </div>
                      </div>
                    </div>
                  }
                >
                  <Touchable
                    onPress={[Function]}
                  >
                    <Touchable
                      onPress={[Function]}
                    >
                      <View
                        onContextMenu={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                      >
                        <div
                          className="amd-guide-modal basic-guide-modal"
                          onContextMenu={[Function]}
                          onTouchCancel={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                        >
                          <Step
                            activeStep={0}
                            goNext={[Function]}
                            goSkip={[Function]}
                            key=".0"
                            showSkip={true}
                            stepIndex={0}
                            stepLen={2}
                          >
                            <div
                              className=""
                            >
                              <div
                                className="top-img"
                              >
                                <img
                                  src="https://gw.alipayobjects.com/mdn/rms_2cf5b1/afts/img/A*DRWiTa03CvwAAAAAAAAAAABkARQnAQ"
                                />
                              </div>
                              <div
                                className="top guide-desc"
                              >
                                <div>
                                  <span
                                    className="blue"
                                  >
                                    运动
                                  </span>
                                  联合
                                  <span
                                    className="yellow"
                                  >
                                    相互宝
                                  </span>
                                </div>
                                <div>
                                  送你健康保障福利
                                </div>
                              </div>
                              <Next
                                goNext={[Function]}
                                goSkip={[Function]}
                                showSkip={true}
                                stepIndex={0}
                                stepLen={2}
                              >
                                <div
                                  className="amd-guide-modal-next-btn-wrap"
                                >
                                  <Button
                                    capsule={true}
                                    className="amd-guide-modal-next-btn"
                                    onPress={[Function]}
                                  >
                                    <ErrorBoundary
                                      onError={[Function]}
                                    >
                                      <Button
                                        activeClassName=""
                                        capsule={true}
                                        capsuleAutoWidth={false}
                                        capsuleSize="md"
                                        className="amd-guide-modal-next-btn"
                                        disabled={false}
                                        loading={false}
                                        onPress={[Function]}
                                        subTitle=""
                                        type="default"
                                      >
                                        <Touchable
                                          disabled={false}
                                          onPress={[Function]}
                                          onPressIn={[Function]}
                                          onPressOut={[Function]}
                                        >
                                          <Touchable
                                            disabled={false}
                                            onPress={[Function]}
                                            onPressIn={[Function]}
                                            onPressOut={[Function]}
                                          >
                                            <View
                                              onContextMenu={[Function]}
                                              onResponderGrant={[Function]}
                                              onResponderMove={[Function]}
                                              onResponderRelease={[Function]}
                                              onResponderTerminate={[Function]}
                                              onResponderTerminationRequest={[Function]}
                                              onStartShouldSetResponder={[Function]}
                                            >
                                              <div
                                                aria-disabled={false}
                                                className="amd-button amd-button-default amd-guide-modal-next-btn amd-button-capsule amd-button-capsule-md"
                                                onContextMenu={[Function]}
                                                onTouchCancel={[Function]}
                                                onTouchEnd={[Function]}
                                                onTouchMove={[Function]}
                                                onTouchStart={[Function]}
                                                role="button"
                                              >
                                                <span
                                                  style={
                                                    Object {
                                                      "opacity": 1,
                                                    }
                                                  }
                                                >
                                                  跳过
                                                </span>
                                              </div>
                                            </View>
                                          </Touchable>
                                        </Touchable>
                                      </Button>
                                    </ErrorBoundary>
                                  </Button>
                                  <Button
                                    capsule={true}
                                    className=""
                                    onPress={[Function]}
                                  >
                                    <ErrorBoundary
                                      onError={[Function]}
                                    >
                                      <Button
                                        activeClassName=""
                                        capsule={true}
                                        capsuleAutoWidth={false}
                                        capsuleSize="md"
                                        className=""
                                        disabled={false}
                                        loading={false}
                                        onPress={[Function]}
                                        subTitle=""
                                        type="default"
                                      >
                                        <Touchable
                                          disabled={false}
                                          onPress={[Function]}
                                          onPressIn={[Function]}
                                          onPressOut={[Function]}
                                        >
                                          <Touchable
                                            disabled={false}
                                            onPress={[Function]}
                                            onPressIn={[Function]}
                                            onPressOut={[Function]}
                                          >
                                            <View
                                              onContextMenu={[Function]}
                                              onResponderGrant={[Function]}
                                              onResponderMove={[Function]}
                                              onResponderRelease={[Function]}
                                              onResponderTerminate={[Function]}
                                              onResponderTerminationRequest={[Function]}
                                              onStartShouldSetResponder={[Function]}
                                            >
                                              <div
                                                aria-disabled={false}
                                                className="amd-button amd-button-default amd-button-capsule amd-button-capsule-md"
                                                onContextMenu={[Function]}
                                                onTouchCancel={[Function]}
                                                onTouchEnd={[Function]}
                                                onTouchMove={[Function]}
                                                onTouchStart={[Function]}
                                                role="button"
                                              >
                                                <span
                                                  style={
                                                    Object {
                                                      "opacity": 1,
                                                    }
                                                  }
                                                >
                                                  下一步
                                                </span>
                                              </div>
                                            </View>
                                          </Touchable>
                                        </Touchable>
                                      </Button>
                                    </ErrorBoundary>
                                  </Button>
                                </div>
                              </Next>
                            </div>
                          </Step>
                          <Step
                            activeStep={0}
                            goNext={[Function]}
                            goSkip={[Function]}
                            key=".1"
                            showSkip={true}
                            stepIndex={1}
                            stepLen={2}
                          >
                            <div
                              className="amd-guide-modal-step-hide"
                            >
                              <div
                                className="bottom-img"
                              >
                                <img
                                  src="https://gw.alipayobjects.com/mdn/rms_2cf5b1/afts/img/A*DRWiTa03CvwAAAAAAAAAAABkARQnAQ"
                                />
                              </div>
                              <div
                                className="bottom guide-desc"
                              >
                                <div>
                                  每天走路领保额，每天最高领
                                  <span
                                    className="yellow"
                                  >
                                    50元
                                  </span>
                                </div>
                                <div>
                                  保障100种重大疾病
                                </div>
                              </div>
                              <Next
                                goNext={[Function]}
                                goSkip={[Function]}
                                showSkip={true}
                                stepIndex={1}
                                stepLen={2}
                              >
                                <div
                                  className="amd-guide-modal-next-btn-wrap"
                                >
                                  <Button
                                    capsule={true}
                                    className="amd-guide-modal-next-btn"
                                    onPress={[Function]}
                                  >
                                    <ErrorBoundary
                                      onError={[Function]}
                                    >
                                      <Button
                                        activeClassName=""
                                        capsule={true}
                                        capsuleAutoWidth={false}
                                        capsuleSize="md"
                                        className="amd-guide-modal-next-btn"
                                        disabled={false}
                                        loading={false}
                                        onPress={[Function]}
                                        subTitle=""
                                        type="default"
                                      >
                                        <Touchable
                                          disabled={false}
                                          onPress={[Function]}
                                          onPressIn={[Function]}
                                          onPressOut={[Function]}
                                        >
                                          <Touchable
                                            disabled={false}
                                            onPress={[Function]}
                                            onPressIn={[Function]}
                                            onPressOut={[Function]}
                                          >
                                            <View
                                              onContextMenu={[Function]}
                                              onResponderGrant={[Function]}
                                              onResponderMove={[Function]}
                                              onResponderRelease={[Function]}
                                              onResponderTerminate={[Function]}
                                              onResponderTerminationRequest={[Function]}
                                              onStartShouldSetResponder={[Function]}
                                            >
                                              <div
                                                aria-disabled={false}
                                                className="amd-button amd-button-default amd-guide-modal-next-btn amd-button-capsule amd-button-capsule-md"
                                                onContextMenu={[Function]}
                                                onTouchCancel={[Function]}
                                                onTouchEnd={[Function]}
                                                onTouchMove={[Function]}
                                                onTouchStart={[Function]}
                                                role="button"
                                              >
                                                <span
                                                  style={
                                                    Object {
                                                      "opacity": 1,
                                                    }
                                                  }
                                                >
                                                  知道了
                                                </span>
                                              </div>
                                            </View>
                                          </Touchable>
                                        </Touchable>
                                      </Button>
                                    </ErrorBoundary>
                                  </Button>
                                </div>
                              </Next>
                            </div>
                          </Step>
                          <div
                            className="amd-guide-modal-indicator-area"
                          >
                            <span
                              className="amd-guide-modal-indicator amd-guide-modal-indicator-active"
                              key="0"
                            />
                            <span
                              className="amd-guide-modal-indicator"
                              key="1"
                            />
                          </div>
                        </div>
                      </View>
                    </Touchable>
                  </Touchable>
                </Portal>
              </GuideModal>
            </ErrorBoundary>
          </GuideModal>
        </div>
      </NewUserGuide>
    </div>
  </_default>
</_default>
`;
